lib/ImportSource.php

```diff
@@ -96,18 +98,33 @@ public function copyToFolder(Folder $destination, string $sourcePath): bool {
 		try {
 			foreach ($files as $path) {
 				if (str_ends_with($path, '/')) {
-					if ($this->copyToFolder($destination->newFolder($path), $sourcePath.$path) === false) {
+					try {
+						$folder = $destination->get($path);
+						if (!($folder instanceof Folder)) {
+							$folder->delete();
+							$folder = $destination->newFolder($path);
+						}
+					} catch (NotFoundException $e) {
+						$folder = $destination->newFolder($path);
+					}
+					if ($this->copyToFolder($folder, $sourcePath.$path) === false) {
+						echo "copy to $sourcePath.$path failed\n";
 						return false;
 					}
 				} else {
 					$stream = $this->archive->getStream($sourcePath.$path, 'r');
 					if ($stream === false) {
 						return false;
 					}
-					$destination->newFile($path, $stream);
+					try {
+						$file = $destination->get($path);
```

Comment:
Should add the correct `$file` typing so `putContent()` doesn't show as undefined